crates/gitbutler-repo/tests/hooks.rs

```diff
@@ -0,0 +1,163 @@
+use gitbutler_repo::hooks;
+
+#[cfg(test)]
+mod tests {
+
+    use std::{
+        collections::HashMap,
+        path::{Path, PathBuf},
+    };
+
+    use gitbutler_repo::staging::{filter_diff_by_hunk_ids, HunkId};
+    use gitbutler_testsupport::{Case, Suite};
+    use hooks::{ErrorData, HookResult, MessageData};
+
+    use super::*;
+
+    #[test]
+    fn pre_commit_hook_success() -> anyhow::Result<()> {
+        let suite = Suite::default();
+        let Case { ctx, .. } = &suite.new_case();
+
+        let selected_hunks = vec![];
+        let hook = b"
+#!/bin/sh
+# do nothing
+";
+        git2_hooks::create_hook(ctx.repo(), git2_hooks::HOOK_PRE_COMMIT, hook);
+        assert_eq!(
+            hooks::pre_commit(ctx, &selected_hunks)?,
+            HookResult::Success
+        );
+        Ok(())
+    }
+
```

Comment:
Can we have an assertion to an existing test or a new test ensuring that the index is brought back to it's original state after `hooks::pre_commit` has returned in both the success and failure situations?

Reply:
Done!